documentation-support/src/main/java/tech/picnic/errorprone/documentation/BugPatternTestExtractor.java

```diff
@@ -0,0 +1,230 @@
+package tech.picnic.errorprone.documentation;
+
+import static com.google.errorprone.matchers.Matchers.instanceMethod;
+import static com.google.errorprone.matchers.method.MethodMatchers.staticMethod;
+import static java.util.function.Predicate.not;
+
+import com.google.auto.service.AutoService;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.annotations.Immutable;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.util.TreeScanner;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.jspecify.annotations.Nullable;
+import tech.picnic.errorprone.documentation.BugPatternTestExtractor.TestCases;
+
+/**
+ * An {@link Extractor} that describes how to extract data from classes that test a {@code
+ * BugChecker}.
+ */
+// XXX: Consider whether to omit or handle differently identification tests without `// BUG:
+// Diagnostic (contains|matches)` markers.
```

Comment:
I think this makes sense. Likewise for unchanged code. (Now that I say this, I vaguely recall that perhaps we discussed pushing this to the website generation script. But I already updated the code :grimacing:. If we'd like to keep this, then more elegant would be to introduce "this code is unchanged" output. That would require further work, though, as it'd require explaining things like [this](https://github.com/PicnicSupermarket/error-prone-support/blob/6d27e46baf64b51d428f75f67277cbff7e39964d/error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/CollectorMutabilityTest.java#L63C1-L78).)